src/porepy/grids/mdg_generation.py

```diff
@@ -0,0 +1,751 @@
+"""
+Module containing a function to generate mixed-dimensional grids (mdg). It encapsulates
+different lower-level mdg generation.
+"""
+from __future__ import annotations
+
+import inspect
+from typing import Callable, Literal, Union, get_args
+
+import numpy as np
+
+import porepy as pp
+from porepy.fracs.fracture_network_2d import FractureNetwork2d
+from porepy.fracs.fracture_network_3d import FractureNetwork3d
+
+FractureNetwork = Union[FractureNetwork2d, FractureNetwork3d]
+
+
+def _validate_args_types(
+    grid_type: Literal["simplex", "cartesian", "tensor_grid"],
+    meshing_args: dict,
+    fracture_network: FractureNetwork,
+):
+    """Validates argument types.
+
+    Parameters:
+        grid_type: Type of grid.
+        meshing_args: A ``dict`` with meshing keys depending on each grid_type.
+        fracture_network: fracture network specification.
+
+    Raises:
+        - TypeError: Raises type error messages if:
+            - grid_type is not ``str``
+            - meshing_args is not ``dict[str]``
+            - fracture_network is not ``FractureNetwork2d`` or ``FractureNetwork3d``
+
+    """
+
+    if not isinstance(grid_type, str):
+        raise TypeError("grid_type must be str, not %r" % type(grid_type))
+
+    if not isinstance(meshing_args, dict):
+        raise TypeError("meshing_args must be dict[str], not %r" % type(meshing_args))
```

Comment:
I'm used to `dict` or `dict[key_type, val_type]`. But `dict[str]` seems misleading. Also, It is not actually tested that the keys must be strings, only that `meshing_args` should be a dictionary. I would say you either add a test or change the error message to reflect what is actually being tested, e.g.,

_meshing_args must be dict ..._

Reply:
Agree, I have updated type hints with dict.